tools/editor/editor_settings.cpp

```diff
@@ -1022,6 +1022,23 @@ void EditorSettings::set_optimize_save(bool p_optimize) {
 	optimize_save=p_optimize;
 }
 
+String EditorSettings::get_last_selected_language()
+{
+	Ref<ConfigFile> cf = memnew( ConfigFile );
+	Error err = cf->load(get_project_settings_path().plus_file("project_metadata.cfg"));
+	if (err != OK) {
+		return "";
+	}
+	return static_cast<String>(cf->get_value("script_setup", "last_selected_language"));
+}
+
+void EditorSettings::set_last_selected_language(String p_language)
+{
+	Ref<ConfigFile> cf = memnew( ConfigFile );
+	cf->set_value("script_setup", "last_selected_language", p_language);
+	cf->save(get_project_settings_path().plus_file("project_metadata.cfg"));
```

Comment:
This would override potential other settings that might be added to project_metadata.cfg in the future, no? It would be best to load the file, set the value and save, to keep existing values for other section/key pairs.